src/logging.rs

```diff
@@ -286,16 +286,26 @@ impl JournalStream {
         Self::from_env("JOURNAL_STREAM")
     }
 
+    /// Get the journal stream that would correspond to the given file metadata.
+    ///
+    /// Return a journal stream struct containing the device and inode number of the given file metadata.
+    pub fn from_metadata(metadata: &Metadata) -> Self {
+        Self {
+            device: metadata.st_dev(),
+            inode: metadata.st_ino(),
+        }
+    }
+
     /// Get the journal stream that would correspond to the given file descriptor.
     ///
     /// Return a journal stream struct containing the device and inode number of the given file descriptor.
     pub fn from_fd<F: AsRawFd>(fd: F) -> std::io::Result<Self> {
-        nix::sys::stat::fstat(fd.as_raw_fd())
-            .map(|stat| JournalStream {
-                device: stat.st_dev,
-                inode: stat.st_ino,
-            })
-            .map_err(std::io::Error::from)
+        // SAFETY: We do claim ownership of the file descriptor here, but we moved it back out down below.
```

Comment:
Self-note: we'll be able to drop these `unsafe` bits once https://doc.rust-lang.org/std/os/unix/io/struct.BorrowedFd.html get stabilized. It will require a signature change though.

Reply:
I think changing the signature would be fine; after all we'd have to bump the MSRV too and that's also a breaking change, isn't it?

Reply:
Indeed :+1:
I'm just dumping here some notes to myself so that in the future I can reconstruct what I had in mind, because I'll surely forget that soon :)